[nnapi] add log_softmax

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61378**

Differential Revision: [D29597355](https://our.internmc.facebook.com/intern/diff/D29597355)

IvanKobzarev committed:
[nnapi] add log_softmax

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29597355).

axitkhurana commented on test/test_nnapi.py:
```diff
@@ -227,6 +227,11 @@ def test_hardtanh(self):
             self.check(torch.nn.Hardtanh(0.0, 5.0), inp)
 
+    def test_log_softmax(self):
+        inp = torch.randn(3, 1000)
```
probably don't need 1000? let's just do a smaller number?

axitkhurana commented on torch/backends/_nnapi/serializer.py:
```diff
@@ -1573,6 +1575,30 @@ def add_conv_underscore(self, node):
+
+        inputs = [None] * 3
+        inputs[0] = input_id
+        inputs[1] = self.add_immediate_float_scalar(1)
```
Let's add a comment on what this 1 is? "specifying the positive scaling factor for the exponent, beta."

axitkhurana reviewed (approved):
Looks good otherwise! Thanks for adding the op

IvanKobzarev committed:
Update on "[nnapi] add log_softmax"


Differential Revision: [D29597355](https://our.internmc.facebook.com/intern/diff/D29597355)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29597355).

IvanKobzarev committed:
Update on "[nnapi] add log_softmax"


Differential Revision: [D29597355](https://our.internmc.facebook.com/intern/diff/D29597355)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29597355).
